src/connection.js

```diff
@@ -1638,8 +1667,23 @@ class Connection extends EventEmitter {
       this.request.rows = [];
       this.request.rst = [];
 
+      let message;
+
+      this.request.once('cancel', () => {
```

Comment:
Somehow there should be a check whether the cancel event stems from the currently active request. When the user calls calcel on an old request or BulkLoad, it should be ignored.

Reply:
You're right! I'll rework this a bit to ensure we only ever handle `cancel` events from the current request. I'll also add some tests for this.